### PR TITLE
PR to merge reddit-model into develop

### PR DESCRIPTION
**Pull request by UmairAhmad125**
In this PR we build the Reddit model that contains
- User with validated attributes
- User can post and the post has all validations
- Post has many comment

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  validates :comment_text, presence: true
```
Comment (lucasmazo32) [inline app/models/comment.rb:5]: - [x] Kindly add validation to check if the input is not empty. For example, a comment like '' should not be valid.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  validates :title, presence: true, uniqueness: { case_sensitive: false }
+  validates :body, presence: true
```
Comment (lucasmazo32) [inline app/models/post.rb:6]: - [x] Kindly check if the body and title are not blank. Currently, I can have a title and body like `tittle: '', body: ''`.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'bcrypt'
+
+class User < ApplicationRecord
+  include BCrypt
+
+  has_many :posts
+  before_save :secure_password
+
+  validates :username, presence: true, uniqueness: { case_sensitive: false },
+                       length: { minimum: 4, maximum: 12 }
```
Comment (lucasmazo32) [inline app/models/user.rb:5]: - [OPTIONAL] You can use length: { in 4..12 } to tell your model to accept any input from length 4 to 12.

**Review by lucasmazo32 — CHANGES_REQUESTED** [review]
# Changes required :wrench:

You are doing a great job so far, and the project is looking good. 

For the project to get approved, kindly implement the changes mentioned below.

### Happy coding! :computer:

## Lucas Mazo

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team ,  you have done a brilliant job with this project. :clap:
#### You can merge  !!!

**Pull request merged**